### ISSUE TITLE
USB connection failed and AppLauncher has stopped

### ISSUE THREAD

**Issue by dy11402** [HappyZ/dpt-tools#172]
Thanks so much for your excellent work! 
I was able to get the root access and modified DPLauncher half a year ago, thanks to your tutorials.
Today, I was trying to install a 3rd party android app and found I cannot connect to DPT through USB. My DPT kept restarting after I connect it to my Mac Pro and there were no devices after I typed `adb devices`. A strange thing is that I can still use your python tool and `fw` function. 
Thus, I repeated the rooting steps again and now the device threw the errors of 'Settings stopped' and 'AppLauncher stopped' in the normal mode....
To sum up, now my device cannot connect via USB (still keep restarting if I plug in the USB cable) and its home button does not work (I guess it's because AppLauncher failed?). 
However, the dpt-tools.py tool with `fw` and Digital Paper App (through wifi connection) still works.
Could you tell me what should I do? Thanks!

**Comment by dy11402** [comment]
I used another Macbook that did not require the USB type a to USB type c connector and problem solved...

**Comment by vznoopy** [comment]
> 非常感谢您的出色工作！
> 半年前，由于您的教程，我能够获得root访问权并修改了DPLauncher。
> 今天，我试图安装第3方android应用程序，发现我无法通过USB连接到DPT。将DPT连接到Mac Pro后，DPT一直重新启动，并且键入后没有任何设备`adb devices`。奇怪的是，我仍然可以使用您的python工具和`fw`功能。
> 因此，我再次重复了生根步骤，现在该设备在正常模式下抛出了“设置已停止”和“ AppLauncher停止”的错误...。
> 总而言之，现在我的设备无法通过USB连接（如果我仍然继续重启，插入USB电缆），并且其主页按钮不起作用（我猜这是因为AppLauncher失败了吗？）。
> 但是，`fw`与Digital Paper App 一起使用的dpt-tools.py工具（通过wifi连接）仍然可以使用。
> 你能告诉我该怎么办吗？谢谢！

In fact, I also encountered the same problem. But I don't have another macbook to fix it. Is there any other solution to this situation?

**Comment by HappyZ** [comment]
Macbook is known to have issues with adb. Have to check if driver is crashed. Ideally if you just log off and log back in, it should work... Also check if your usb-c cable is for charging only..

**Comment by vznoopy** [comment]
> Macbook is known to have issues with adb. Have to check if driver is crashed. Ideally if you just log off and log back in, it should work... Also check if your usb-c cable is for charging only..

Thank you for your reply. I have tried many reboots and tried various cable. But it did not solve the problem.

The key is the RP1 kept restarting after connect it to Macbook Pro.
